### PR TITLE
Alex/#13 implement minio for object storage

### LINKED ISSUE #13: Implement Minio for object storage
- [x] Add Minio to docker-compose
- [x] Configure application to download image and upload to our object storage
- [x] save link in DB of our object storage.
- [x] Make sure all configurations are done as ENV in docker-compose.

### PR DESCRIPTION
**Pull request by aleksandr-efimenko**
Implemented functionality to store generated images in an S3-compatible MinIO bucket. 
The MinIO service is configured to run with docker-compose.
When a user opens the image generation page they will see the last 9 generated images.
While images are loading users see animated blocks.

Closes #13

### CODE REVIEW

File: next.config.mjs [hunk 1]
```diff
@@ -32,6 +42,13 @@ const config = {
         pathname: "/**/*",
         port: "",
       },
+      // for minio images
+      {
+        hostname: "minio",
```
Comment (cohenoshriconsulting) [inline next.config.mjs:47]: Move the hostname to environment variable

File: next.config.mjs [hunk 2]
```diff
@@ -32,6 +42,13 @@ const config = {
...
+        protocol: "http",
+        pathname: "/**/*",
+        port: "9000",
```
Comment (cohenoshriconsulting) [inline next.config.mjs:50]: Don't forget about the port

File: package.json [hunk 3]
```diff
@@ -34,11 +34,13 @@
     "@trpc/react-query": "^10.34.0",
     "@trpc/server": "^10.34.0",
     "ai": "^2.2.6",
+    "axios": "^1.5.1",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.0.0",
     "cmdk": "^0.2.0",
     "i18next": "^23.5.1",
     "lucide-react": "^0.268.0",
+    "minio": "^7.1.3",
```
Comment (cohenoshriconsulting) [inline package.json:43]: remove the minio package

File: prisma/schema.prisma [hunk 4]
```diff
@@ -99,7 +99,8 @@ model ImageGenerationRequest {
 model GeneratedImages {
     id                       String                  @id @default(cuid())
     createdAt                DateTime                @default(now())
-    imageUrl                 String
+    imageName                String
+    imageBucketName          String
```
Comment (cohenoshriconsulting) [inline prisma/schema.prisma:103]: Only store the final public URL from our object storage bucket

File: src/env.mjs [hunk 5]
```diff
@@ -54,6 +63,16 @@ export const env = createEnv({
...
+
+    MINIO_ACCESS_KEY: process.env.MINIO_ACCESS_KEY,
+    MINIO_SECRET_KEY: process.env.MINIO_SECRET_KEY,
+    MINIO_DISTRIBUTED_MODE_ENABLED:
+      process.env.MINIO_DISTRIBUTED_MODE_ENABLED === "true",
+    MINIO_BUCKET_NAME: process.env.MINIO_BUCKET_NAME,
+
+    MINIO_ENDPOINT: process.env.MINIO_ENDPOINT,
+    MINIO_PORT: Number(process.env.MINIO_PORT),
+    MINIO_USE_SSL: process.env.MINIO_USE_SSL === "true",
```
Comment (cohenoshriconsulting) [inline src/env.mjs:75]: Remove minio client as it is only for development

**Review by cohenoshriconsulting — APPROVED** [review]

File: next.config.mjs (hunk 1 above)
Comment (aleksandr-efimenko) [reply next.config.mjs]: Add S3_ENDPOINT env variable

**Pull request merged**